examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,578 @@
+    dashes=False, palette='Set1', marker='o', alpha=.5, ax=ax
+)
+ax.set_xlabel("CV test fold", size=12, labelpad=10)
+ax.set_ylabel("Model AUC", size=12)
+ax.tick_params(bottom=True, labelbottom=False)
+plt.show()
+
+# print correlation of AUC scores across folds
+print(f"Correlation of models:\n {model_scores.transpose().corr()}")
+
+# %%
+# We can observe that the performance of the models highly depends on the fold.
+#
+# As a consequence, if we assume independence between observations we will be
+# underestimating the variance computed in our statistical tests, increasing
+# the number of false positive errors (i.e. detecting a significant difference
+# between models when such does not exist) [1]_.
+#
+# Several methods have been developed to correct the variance for the
+# correlation in these cases. In this example we will explore one of these
+# methods implemented using two different statistical approaches: frequentist
+# and Bayesian.
+
+# %%
+# Comparing two models: frequentist approach
+# ------------------------------------------
+#
+# We can start by asking: "Is the first model significantly better than the
+# second model (when ranked by `mean_test_score`)?"
+#
+# To answer this question using a frequentist approach we could
+# run a paired t-test and compute the p-value. This is also known as
+# Diebold-Mariano test in the forecast literature [5]_.
+# Many variants of the latter have been developed to account for the
+# 'non-independence of observations problem'
+# described in the previous section. We will use the one proven to obtain the
+# highest replicability scores (which rate how similar the performance of a
+# model is when evaluating it on different random partitions of the same
+# dataset) while mantaining a low rate of false postitives and false negatives:
+# the Nadeau and Bengio's corrected t-test [2]_ that uses a 10 times repeated
+# 10-fold cross validation [3]_.
+#
+# This corrected paired t-test is computed as:
+#
+# .. math::
+#    t=\frac{\frac{1}{k \cdot r}\sum_{i=1}^{k}\sum_{j=1}^{r}x_{ij}}
+#    {\sqrt{(\frac{1}{k \cdot r}+\frac{n_{test}}{n_{train}})\hat{\sigma}^2}}
+#
+# where :math:`k` is the number of folds,
+# :math:`r` the number of repetitions in the cross-validation,
+# :math:`x` is the difference in performance of the models,
+# :math:`n_{test}` is the number of observations used for testing,
+# :math:`n_{train}` is the the number of observations used for training,
+# and :math:`\hat{\sigma}^2` represents the variance of the sample.
+#
+# Let's implement a corrected right-tailed paired t-test to evaluate if the
+# performance of the first model is significantly better than that of the
+# second model. Our null hypothesis is that the second model performs at least
+# as good as the first model.
+
+import numpy as np
+from scipy.stats import t
+
+
+def correct_std(differences, n, n_train, n_test):
+    """Corrects standard deviation using Nadeau and Bengio's approach.
+
+    Parameters
```

Comment:
I added these docstrings in response to @glemaitre 's comment [here](https://github.com/scikit-learn/scikit-learn/pull/17432#discussion_r440721828).  I don't have a strong preference on this (except that maybe it will be easier to keep it as it is now), but if one of you has a strong preference I'm happy to go with that one.